olive/passes/onnx/conversion.py

```diff
@@ -124,10 +133,11 @@ def _run_for_config(
             onnx_models.append(output_model_component)
             component_names.append(component_name)
 
-        if onnx_models:
-            return CompositeModelHandler(onnx_models, component_names)
-        else:
-            return self._convert_model_on_device(model, data_root, config, output_model_path, device, torch_dtype)
+        if config["merge_components"] and len(onnx_models) == 2:
+            merged_model = merge_decoders(onnx_models[0].model_path, onnx_models[1].model_path)
+            return model_proto_to_olive_model(merged_model, output_model_path, config)
```

Comment:
Should we delete the component sub directories since they are already merged? The components were also saved in the output model directory. 

Might also have to resolve the final `output_model_path` like at https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/conversion.py#L375. Otherwise, the onnx model will probably be saved without a `.onnx` suffix.

Reply:
I don't think we should delete the component models. The component models wouldn't end up in the output folder unless this is the last pass in the config. Even in that case, the component models aren't returned so they shouldn't be in the output folder.

Fixed the path resolve issue.